Tom Rafferty - tv-show-dom-project - version 1.9 - level 300

TomRafferty opened the pull request:
This is a review version of the tv-show-dom-project.
This version should hold all necessary functionality for level 300 and below.
CSS is basic and completely unfinished.

TomRafferty committed:
added my name and my github username

TomRafferty committed:
Pseudo added for cards + got rid of some unneeded code.

this will currently display nothing on the page and simply serves as a base for the build of the site.

TomRafferty committed:
Version 1.0 - core

The base of the website needed for level 100 of the project is down. I have added a basic forEach() loop that handles each episode as cards and will now overlay some CSS to make the page more readable and enjoyable to look at.

TomRafferty committed:
Version 1.1 - CSS

added base for css - building in place for multiple devices.

TomRafferty committed:
Version 1.2 - Working on search bar

currently the live search is broken. I can't understand the results I'm getting it just seems to remove some and delete others while also cloning some. Currently trying to fix.

TomRafferty committed:
version 1.3

updating remote

TomRafferty committed:
version 1.4 - basic live search

live search has been successfully implemented. After Nick pointed out that there were a few small but crucial errors. I was able to fix them. Now just to remove the case sensitivity.

TomRafferty committed:
version 1.4 - fully implemented live search

Live search now works great, case sensitivity is no longer an issue.

TomRafferty committed:
version 1.5 - level 200 completed

now with live search fully implemented I finished off level 200 with the result counter which appears only when there is a search term in place.

TomRafferty committed:
version 1.6 - minor result count change

changed the results counter to give a slight bit more information - matching results/total.

TomRafferty committed:
version 1.7 - level 300 complete - select function

select function fully implemented. works alongside the live search function with no flaws (that I've noticed).

TomRafferty committed:
version 1.8 - minor bug fix

bug squashed - some summaries still had element tags in them so with a slight alteration of the code it now works fine.

TomRafferty committed:
version 1.9 - colour changes and slight css changes

just tweaked some stuff inside the style sheet

Haseeb-Khalil commented on script.js:
```diff
@@ -28,6 +30,46 @@ function makePageForEpisodes(episodeList) {
+    cardDiv.appendChild(headingEl);
+    //create subheading (season number and episode code)
+    const subHeadingEl = document.createElement("h3");
+    subHeadingEl.innerHTML = `SO${episode.season}EP${episode.number}`;
```
very well intended code Tom
I believe you have to add zero before season & episode number with condition.
i.e if(episode.season < 10)

Haseeb-Khalil commented:
Very well intended code Tom
I like the use of comments. Your code is simple and easy to understand 👍 
Keep it up. I've mentioned in a line comment where you might want to amend your code. Apart from that 🥇

TomRafferty committed:
partly functioning live data

broken select functionality

TomRafferty committed:
fixed select bug - level 350 complete

live data is now implemented and I believe all aspects of the site are currently working

TomRafferty committed:
fixed number formatting

season and episode numbers a now formatted correctly.

TomRafferty committed:
started switching to DOM over HTML (BROKEN BUILD)

This build does not work.
I'm unsure why as it should just be running the code as it was when I was using hardcoded HTML

TomRafferty committed:
version 2 - a new attempt

my old method wasn't working and I have had to revert a lot of my code to its old functioning state. However it dawned on me that I was creating a bigger issue for myself than needed.

TomRafferty committed:
version 2.1 - broken master

master... the code is broken!
but it's ok, the code gods have praised you with the ability to log a version and refactor!

TomRafferty committed:
added select show element

added the element in for the show selector and imported the hard coded data into a variable - allShows

TomRafferty committed:
async data fetch setup

the data is now retrieved properly using async await - also switched allShows to live data

TomRafferty committed:
select show partially implemented

Select show doesn't select a specific show. That's next.

TomRafferty committed:
Error - type error

I feel like this is just a silly problem. All the code that handles the main functionality has worked previously, I believe I'm just calling something to early but I can't figure out what. I'm hoping to get a fresh perspective on the code.

TomRafferty committed:
semi-functional

this commit is semi-functional all shows will load, however the first time you load a show will throw an error and if you try to search using the live search function of the select episode function it just throws more errors and breaks the site.

TomRafferty committed:
Level 400 - All known issues fixed

I believe I have met the goals of level 400.
Finally!
Now the shows selector is in alphabetical order and everything works without errors.

TomRafferty committed:
refactor complete

I have made the code as readable as I can for now and I believe it is acceptable.
Now I work work on polishing the project a little and if I have time I will move onto level 500

TomRafferty committed:
Merge branch 'Full-Refactor'
